main.js

```diff
@@ -35,6 +35,37 @@ function runOpenAndClose(property) {
   mobileMenu.style.display = property;
 }
 
+function onlineLink(){
+  const onlines = document.querySelectorAll(".Online");
+  console.log(onlines.length)
+  onlines.forEach(button =>{
+    button.addEventListener('click', () => {
+      window.location.replace('/filter.htm?online')
+
+    })
+  });
+  console.log('online links done')
+}
+onlineLink()
+
+function onsiteLink(){
+  const onsites = document.querySelectorAll(".OnSite");
+  console.log(onsites.length)
+  onsites.forEach(button => {
+    button.addEventListener('click', () => {
+      window.location.replace('/filter.htm?onsite')
+
+    })
+    
+  });
+  console.log('onsite links done')
+}
+onsiteLink();
```

Comment:
This seems unnecessarily complex to me. The query string links (in HTML) should work fine, require less code, provide the same user experience and be less error prone – all in all a better choice over adding this extra code.